### PR TITLE
Map CLI exit codes by scanner type (AST-37829)

### PR DESCRIPTION
**Pull request by checkmarx-kobi-hagmi**
### Description

> Apply specific CLI exit codes to indicate which of the scanners failed during a scan.

### References

> https://checkmarx.atlassian.net/browse/AST-37829

### Testing

> Added unit tests for one failing scanner, multiple fails and partial scan.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

**Review by margaritalm — APPROVED** [review]
The code good. Maybe need to update AST-37829 with 
	SastExitCode   = 2
	ScaExitCode    = 3
	IacExitCode    = 4
	KicsExitCode   = 5
	ApisecExitCode = 6

File: internal/wrappers/mock/projects-mock.go [hunk 1]
```diff
@@ -40,6 +40,54 @@ func (p *ProjectsMockWrapper) Get(params map[string]string) (
 		filteredTotalCount = 0
 	}
 
+	if params["names"] == "fake-kics-scanner-fail" {
```
Comment (OrShamirCM) [inline internal/wrappers/mock/projects-mock.go:43]: Suggestion -you can change all this makes to switch case and a method the creates the fake with params

File: internal/errors/exit-codes/exit-codes.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package exitcodes
+
+const (
+	GeneralExitCode = 1
+	SastExitCode    = 2
```
Comment (OrShamirCM) [inline internal/errors/exit-codes/exit-codes.go:5]: Sast/engineFailedExitCode

File: internal/errors/exit-codes/exit-codes.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package exitcodes
+
+const (
+	GeneralExitCode = 1
```
Comment (OrShamirCM) [inline internal/errors/exit-codes/exit-codes.go:4]: MultipleEngineFailureExitsCode
Please use more meaningful names

File: internal/commands/scan.go [hunk 4]
```diff
@@ -1992,13 +1992,49 @@ func isScanRunning(
...
 	}
 	return false, nil
 }
 
+func getExitCode(scanResponseModel *wrappers.ScanResponseModel) int {
+	failedStatuses := make([]int, 0)
+	for _, scanner := range scanResponseModel.StatusDetails {
+		scannerNameLowerCase := strings.ToLower(scanner.Name)
+		scannerErrorExitCode, errorCodeByScannerExists := errorCodesByScanner[scannerNameLowerCase]
+		if scanner.Status == wrappers.ScanFailed && scanner.Name != General && errorCodeByScannerExists {
```
Comment (OrShamirCM) [inline internal/commands/scan.go:2009]: make sure the only status we need is failed

Comment (checkmarx-kobi-hagmi) [reply internal/commands/scan.go:2017]: Also partial. verified

Comment (OrShamirCM) [reply internal/commands/scan.go:2017]: did you change it?

File: internal/commands/scan.go [hunk 5]
```diff
@@ -2000,13 +2001,49 @@ func isScanRunning(
...
+	}
+
+	return exitCodes.MultipleEnginesFailedExitCode
+}
+
+const (
+	General = "general"
+	Sast    = "sast"
+	Sca     = "sca"
+	Iac     = "iac"
```
Comment (checkmarx-kobi-hagmi) [inline internal/commands/scan.go:2033]: check terminology - kics, iac, iac-security

**Review by OrShamirCM — APPROVED** [review]

**Pull request merged**